crates/optimism/chainspec/src/lib.rs

```diff
@@ -32,21 +32,53 @@ use reth_chainspec::{
     Hardforks, Head,
 };
 use reth_network_peers::NodeRecord;
+use reth_optimism_forks::OptimismHardfork;
 use reth_primitives_traits::Header;
 
+const DENOMINATOR_MASK: u64 = 0xFFFFFFFF00000000;
+const ELASTICITY_MASK: u64 = 0x00000000FFFFFFFF;
+
 /// OP stack chain spec type.
 #[derive(Debug, Clone, Deref, Into, Constructor, PartialEq, Eq)]
 pub struct OpChainSpec {
     /// [`ChainSpec`].
     pub inner: ChainSpec,
 }
 
+/// Fee trait for OP chain specs.
+pub trait Fee {
+    /// Read from parent to determine the base fee for the next block
+    fn next_block_base_fee(&self, parent: &Header, timestamp: u64) -> U256;
+}
+
 /// Returns the signature for the optimism deposit transactions, which don't include a
 /// signature.
 pub fn optimism_deposit_tx_signature() -> Signature {
     Signature::new(U256::ZERO, U256::ZERO, Parity::Parity(false))
 }
 
+impl Fee for OpChainSpec {
+    fn next_block_base_fee(&self, parent: &Header, timestamp: u64) -> U256 {
+        let is_holocene =
+            self.inner.is_fork_active_at_timestamp(OptimismHardfork::Holocene, timestamp);
+        if is_holocene {
+            // First 4 bytes of the nonce are the base fee denominator, the last 4 bytes are the
+            // elasticity
+            let denominator = parent.nonce & B64::from_str(&DENOMINATOR_MASK.to_string()).unwrap();
+            let elasticity = parent.nonce & B64::from_str(&ELASTICITY_MASK.to_string()).unwrap();
```

Comment:
Can we remove the `unwrap`s here by making the constants declared as `B64`s using the [`b64!`](https://docs.rs/alloy-primitives/latest/alloy_primitives/macro.b64.html) macro?

e.g.
```rust
const DENOMINATOR_MASK: B64 = b64!("FFFFFFFF00000000");
const ELASTICITY_MASK: B64 = b64!("00000000FFFFFFFF");
```